File: QFinance_Interviews/LeetCode/reverseinteger.cpp
#include<iostream>
#include<deque>

class Solution
{

public:

  // Reverses an integer
  // (ex) 123 becomes 321
  int ReverseInteger(int num)
  {

    int result = 0;

    while (num != 0){
      if (std::abs(num) > 214748364){
	return 0;
      }

      result = result * 10 + num%10;
      num /= 10; 
      
    }

    return result;
  }


};


int main()
{

  int number = -1234;

  Solution sol;
  
  int reversed = sol.ReverseInteger(number);

  std::cout << "Input is " << number << "\n";
  std::cout << "Reversed is " << reversed << "\n";

}

File: QFinance_Interviews/LeetCode/reverseinteger.py
#!/usr/bin/python3

def reverse_integer(num):

    rnum = 0
    negative = False
    
    if num < 0:
        negative = True
        num = abs(num)
    
    while num != 0:

        if (rnum > 214748364):
            return 0
        
        rnum = rnum * 10 + num % 10
        num = num // 10

    if negative == True:
        rnum = -rnum

    return rnum
    
print(reverse_integer(1000000003))
